tuned_lens/plotting/prediction_trajectory.py

```diff
@@ -230,6 +229,43 @@ def largest_prob_labels(
             hover_over_entries=entry_format_fn(topk_tokens, topk_probs * 100),
         )
 
+    def get_first_order_diff(self, k: int = 10) -> Any:
+        # WIP
+        # Pseudocode:
+
+        # for each input token:
+        #     for each layer (starting at 2nd layer):
+        #         get top k tokens for that layer
```

Comment:
This should be vectorized. We don't want to be looping over the token index this will be 40k+ tokens per layer that is going to be slow. Instead, we should use an element wise subtraction. If `layer[i, :]` and `layer[i+1, :]` are `NDArray`s then we should just do `layer[i+1, :] - layer[i, :]`. 

In addition, we should not be thinking about the top k tokens at this point since in the end the tokens that have changed the most may not be among top k tokens in either layer. We may need to do something in the select top k tokens code latter but don't worry about that just yet.